lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -16,15 +16,38 @@
 #include "mlir/IR/DialectImplementation.h"
 #include "mlir/IR/FunctionImplementation.h"
 #include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/SymbolTable.h"
 #include "llvm/ADT/DenseMap.h"
 #include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallSet.h"
 #include "llvm/ADT/StringExtras.h"
 #include "llvm/ADT/TypeSwitch.h"
 
 using namespace circt;
 using namespace circt::calyx;
 using namespace mlir;
 
+LogicalResult calyx::verifyControlLikeOpBody(Operation *op) {
+  auto &region = op->getRegion(0);
+
+  bool isNotControlOp = !isa<ControlOp>(op);
+  for (auto &&bodyOp : region.front()) {
+    if (isa<SeqOp>(bodyOp))
+      continue;
+
+    if (isNotControlOp && isa<EnableOp>(bodyOp))
```

Comment:
isn't this redundant? i.e. if it's an EnableOp, it's not a ControlOp? i assume this is leftover from some refactoring or something. it seems like there could be one

```
if isa<SeqOp, EnableOp>(bodyOp)
  continue;
```

Up top, so theres just one allowlist of acceptable ops to have in ControlLike regions.

Reply:
I was trying to catch the case where the op is an EnableOp, but located inside the ControlOp; I thought this was not allowed. However, I just tried it on the native Calyx compiler and indeed it is fine. 

I've fixed this and moved the allowed ops in a lambda for cleanliness. I'm going to confirm this behavior is indeed correct.